src/v/coproc/pacemaker.cc

```diff
@@ -46,30 +50,44 @@ pacemaker::pacemaker(ss::socket_address addr, ss::sharded<storage::api>& api)
       wasm_transport_cfg(addr), wasm_transport_backoff()),
     api.local()) {}
 
+ss::future<> pacemaker::start() {
+    _offs.timer.set_callback([this] {
+        (void)ss::with_gate(_offs.gate, [this] {
+            return save_offsets(_offs.snap, _ntps).then([this] {
+                _offs.timer.arm(_offs.duration);
+            });
+        });
+    });
+    co_await ss::recursive_touch_directory(offsets_snapshot_path().string());
+    co_await ss::with_gate(_offs.gate, [this] {
```

Comment:
this `with_gate` is suspect. does the caller ever invoke stop before start's future has been set? a general design pattern that i think we tend to do a good job of enforcing is that `foo::start` resolves before any chance of concurrent access to `foo`. if this is an exception to the rule i think it needs a description of the scenario.

Reply:
Nope , your right, I just removed it